File: Python/7py 1/add_letters.py
"""
|--------------------------------------------------------------------------
| Alphabetical Addition
|--------------------------------------------------------------------------
|
"""

def add_letters(*letters):
   abj = "zabcdefghijklmnopqrstuvwxyz"
   return abj[sum([abj.index(v) for v in letters]) % 26]


def add_letters(*letters):
   return chr(96 + (sum(ord(l) - 96 for l in letters) % 26 or 26))


def add_letters(*letters):
   s = sum(122-ord(l) for l in letters)
   return chr(122 - s % 26)


print(add_letters("a", "b", "c"), "f")
print(add_letters("y", "c", "b"), "d")
print(add_letters(), "z")

File: PHP/7php 1/addLetters.php
<?php
/*
|--------------------------------------------------------------------------
| Alphabetical Addition
|--------------------------------------------------------------------------
|
*/

function addLetters(...$letters) {
   $abj = "zabcdefghijklmnopqrstuvwxyz";
   $res = 0;
   foreach ($letters as $v) $res += strpos($abj, $v);
   return $abj[$res%26];
}


function addLetters(...$arr): string {
   if (empty($arr)) return 'z';
   $s = array_reduce($arr, function($carry, $item){
      return $carry + ord($item) - 96;
   }) - 1;
   return chr($s % 26 + 97);
}


function addLetters(...$letters) {
   return !empty($letters) ? chr((array_sum(array_map(function($char){
      return ord($char) - 96;
   }, $letters)) - 1) % 26 + 97) : 'z';
}


print_r(addLetters("a", "b", "c") . " f
");
print_r(addLetters("y", "c", "b") . " d
");
print_r(addLetters() . " z
");